File: src/main/java/com/ragin/bdd/cucumber/datetimeformat/BddCucumberDateTimeFormat.java
package com.ragin.bdd.cucumber.datetimeformat;

import java.time.format.DateTimeFormatter;
import java.util.List;

@FunctionalInterface
public interface BddCucumberDateTimeFormat {
    /**
     * Pattern to add to the Date(Time)Parser.
     *
     * @return List of patterns to add
     */
    List<DateTimeFormatter> formatters();
}

File: src/main/kotlin/com/ragin/bdd/cucumber/datetimeformat/BddCucumberDateTimeFormat.kt
package com.ragin.bdd.cucumber.datetimeformat

import java.time.format.DateTimeFormatter

fun interface BddCucumberDateTimeFormat {
    /**
     * Pattern to add to the Date(Time)Parser.
     *
     * @return List of patterns to add
     */
    fun formatters(): List<DateTimeFormatter>?
}
